src/lib.rs

```diff
@@ -134,6 +283,43 @@ pub trait FutureExt: Future + Sized {
 
 impl<F> FutureExt for F where F: Future + Sized {}
 
+impl<T> FfiPoll<T> {
+    /// Converts a [`std::task::Poll`] to the [`FfiPoll`]
+    ///
+    /// [`std::task::Poll`]: std::task::Poll
+    pub fn from_poll(poll: Poll<T>) -> Self {
+        match poll {
+            Poll::Ready(r) => Self::Ready(r),
+            Poll::Pending => Self::Pending,
+        }
+    }
+    /// Converts a [`FfiPoll`] back to the [`std::task::Poll`]
+    ///
+    /// [`std::task::Poll`]: std::task::Poll
+    pub fn into_poll(self) -> Poll<T> {
+        match self {
+            Self::Ready(r) => Poll::Ready(r),
+            Self::Pending => Poll::Pending,
+            Self::Panicked => {
+                // FFI panicked, so let us panic too.
+                panic!("FFI future unwinded.");
+            }
+        }
+    }
+}
+
+impl<T> From<Poll<T>> for FfiPoll<T> {
+    fn from(poll: Poll<T>) -> Self {
+        Self::from_poll(poll)
+    }
+}
+
+impl<T> From<FfiPoll<T>> for Poll<T> {
+    fn from(ffi_poll: FfiPoll<T>) -> Self {
+        ffi_poll.into_poll()
```

Comment:
According to the doc, `From::from` should not fail.
https://doc.rust-lang.org/stable/std/convert/trait.From.html